corehq/apps/linked_domain/tests/test_update_custom_data_fields.py

```diff
@@ -85,6 +90,189 @@ def test_update(self):
+        existing_definition.set_fields(fields)
+        existing_definition.save()
+        make_existing_profile(existing_definition, is_synced=True)
+
+        user_fields_definition = {
+            'fields': [field_to_json(field) for field in fields],
+            'profiles': [make_profile()]
+        }
+        update_json = {'UserFields': user_fields_definition}
+
+        update_custom_data_models_impl(update_json, 'test-domain')
+
+        updated_definition = CustomDataFieldsDefinition.objects.get(domain='test-domain', field_type='UserFields')
+        profiles = updated_definition.get_profiles()
+        self.assertEqual(len(profiles), 1)
+
+        profile = profiles[0]
+        self.assertEqual(profile.fields, "{'a': 'one'}")
+        self.assertTrue(profile.is_synced)
+
+    def test_existing_profile_raises_error(self):
+        fields = [Field(slug='a', is_synced=True)]
+        existing_definition = CustomDataFieldsDefinition.objects.create(
+            domain='test-domain', field_type='UserFields'
+        )
+        existing_definition.set_fields(fields)
+        existing_definition.save()
+        make_existing_profile(existing_definition, is_synced=False)
+
+        user_fields_definition = {
+            'fields': [field_to_json(field) for field in fields],
+            'profiles': [make_profile()]
+        }
+        update_json = {'UserFields': user_fields_definition}
+
+        with self.assertRaisesMessage(DomainLinkError,
+                'Cannot update custom fields due to the following profile conflicts: SyncedProfile1'):
+            update_custom_data_models_impl(update_json, 'test-domain')
+
+    def test_can_remove_profile(self):
+        fields = [Field(slug='a', is_synced=True)]
+        existing_definition = CustomDataFieldsDefinition.objects.create(
+            domain='test-domain', field_type='UserFields'
+        )
+        existing_definition.set_fields(fields)
+        existing_definition.save()
+        make_existing_profile(existing_definition, is_synced=True)
+
+        user_fields_definition = {
+            'fields': [field_to_json(field) for field in fields],
+            # No Profiles
+        }
+        update_json = {'UserFields': user_fields_definition}
+
+        update_custom_data_models_impl(update_json, 'test-domain')
+
+        updated_definition = CustomDataFieldsDefinition.objects.get(domain='test-domain', field_type='UserFields')
+        self.assertEqual(updated_definition.get_profiles(), [])
+
+    def test_removing_used_profile_does_not_delete_the_profile(self):
+        # NOTE: Initially I tried to have profiles actually assigned to users,
+        # but ran into some issues I believe due to how the test environment interacts with Elasticsearch.
+        # Due to this, I'm just faking out 'has_users_assigned' to always believe that the profile is in use.
+
+        # create a profile
+        fields = [Field(slug='a', is_synced=True), ]
```

Comment:
nit
```suggestion
        fields = [Field(slug='a', is_synced=True)]
```

Reply:
appears to be outdated -- `is_synced` was replaced with `upstream_id`. I'll try to do better about removing historical artifacts in the future.